### ISSUE TITLE
Javascript Error Running on Windows 10 / WSL

### ISSUE THREAD

**Issue by matrixraven** [sedwards2009/extraterm#171]
My setup is Windows 10 latest updates, WSL (default legacy container uninstalled and replaced with Debian downloaded from the Windows Store). The container installs as stretch, so I updated it to buster since that's what I use on my Pixelbook (where I successfully use extraterm as my default terminal app).

I used the Windows installer and while I can launch the app, the cursor just sits there and I never get a prompt. If I open another tab, a huge Javascript error shows in a Windows error popup so I'm guessing that's the issue.

**Comment by sedwards2009** [comment]
Can you post a screenshot of the error message? (Pro Tip: If you focus on the error pop up and press Ctrl+C it may copy all of the contents to the clipboard.) Also, does WSL work if you just use the Windows console?

**Comment by matrixraven** [comment]
[Window Title]
Error

[Main Instruction]
A JavaScript error occurred in the main process

[Content]
Uncaught Exception:
Error [ERR_STREAM_DESTROYED]: Cannot call write after a stream was destroyed
    at doWrite (_stream_writable.js:406:19)
    at writeOrBuffer (_stream_writable.js:394:5)
    at Socket.Writable.write (_stream_writable.js:294:11)
    at WslProxyPtyConnector._sendMessage (E:\Program Files\Extraterm\resources\app\extensions\ProxySessionBackend\src\ProxyPty.js:212:27)
    at WslProxyPtyConnector.spawn (E:\Program Files\Extraterm\resources\app\extensions\ProxySessionBackend\src\ProxyPty.js:141:14)
    at WslProxySessionBackend.createSession (E:\Program Files\Extraterm\resources\app\extensions\ProxySessionBackend\src\WslProxySessionBackend.js:74:26)
    at PtyManager.createPty (E:\Program Files\Extraterm\resources\app\extraterm\src\main_process\pty\PtyManager.js:51:33)
    at handlePtyCreate (E:\Program Files\Extraterm\resources\app\extraterm\src\main_process\Main.js:757:30)
    at EventEmitter.handleIpc (E:\Program Files\Extraterm\resources\app\extraterm\src\main_process\Main.js:549:21)
    at EventEmitter.emit (events.js:182:13)

[OK]

Yes, WSL works just fine when I run it by cilcking the Debian icon in my start menu.

**Comment by sedwards2009** [comment]
Do you by any chance have some kind of anti-virus software installed?

**Comment by matrixraven** [comment]
Malwarebytes Pro and the built in Windows Defender. However, I completely exited Malwarebytes and it doesn't change the error. Any other ideas?

**Comment by jefftimlin** [comment]
I started getting this behaviour after setting up a wsl.conf file as described here: https://docs.microsoft.com/en-us/windows/wsl/wsl-config#set-wsl-launch-settings

Removing the wsl.conf file fixes the issue.

**Comment by sedwards2009** [comment]
I've got a fix queued up for the next bug fix release which makes Extraterm not fall over if the Windowsdrives are mounted anywhere other than `/mnt/c/`. This might fix the problems reported here. Also Extraterm assumes python3 is available on the Linux side.

This complex situation should improve in the future once ConPTY is more mature.

**Comment by seffyroff** [comment]
I'm seeing the same thing, after installing extraterm, then installing Fish and doing chsh -s /usr/bin/fish.  No WSL shells work after doing this - tried Fish, Bash, ZSH.  

Here's the error message for completeness:
`[Window Title]
Error

[Main Instruction]
A JavaScript error occurred in the main process

[Content]
Uncaught Exception:
Error [ERR_STREAM_DESTROYED]: Cannot call write after a stream was destroyed
    at doWrite (_stream_writable.js:406:19)
    at writeOrBuffer (_stream_writable.js:394:5)
    at Socket.Writable.write (_stream_writable.js:294:11)
    at WslProxyPtyConnector._sendMessage (C:\Program Files\Extraterm\resources\app\extensions\ProxySessionBackend\src\ProxyPty.js:212:27)
    at WslProxyPtyConnector.spawn (C:\Program Files\Extraterm\resources\app\extensions\ProxySessionBackend\src\ProxyPty.js:141:14)
    at WslProxySessionBackend.createSession (C:\Program Files\Extraterm\resources\app\extensions\ProxySessionBackend\src\WslProxySessionBackend.js:74:26)
    at PtyManager.createPty (C:\Program Files\Extraterm\resources\app\extraterm\src\main_process\pty\PtyManager.js:51:33)
    at handlePtyCreate (C:\Program Files\Extraterm\resources\app\extraterm\src\main_process\Main.js:838:30)
    at EventEmitter.handleIpc (C:\Program Files\Extraterm\resources\app\extraterm\src\main_process\Main.js:630:21)
    at EventEmitter.emit (events.js:182:13)

[OK]`
Python 2 and 3 are installed in WSL and i have no custom /etc/wsl.conf.

**Comment by philmerricksowi** [comment]
I tried this out on another (cleaner) Win10 box today, and see the same issue.

**Comment by lewenweijia** [comment]
`chsh -s /usr/bin/fish` will lead to the same issue to me, while `chsh -s /usr/bin/zsh` make ExtraTerm work fine again. Now I choose the prefered shell in Settings=>Session Types=>WSL=>Shell/other, keep chsh unchange.

**Comment by Kynolin** [comment]
A workaround I use is to change the instance shell to other (in Extraterm) and specify /usr/bin/fish, leaving the user's default shell as /bin/bash (in WSL).

When I first started using WSL, changing the user's default shell to fish would cause the WSL instance to never get to a prompt (after all instances were closed and a fresh one started), regardless of the application I used to launch it. I'm still on Windows 10 version 1709 at work, and I'm currently forced to use the legacy lxss version, so I went straight for the workaround.

I haven't tested the behavior on newer versions of WSL from the Windows store at home.

**Comment by matrixraven** [comment]
Sorry for getting back so late. I've updated and it seems to run fine now. I think the issue was the WSL mount thing that you fixed a few point releases ago. I do have my Extraterm installed on E drive, so maybe that was the issue. Is it ok to close this now?

**Comment by sedwards2009** [comment]
@matrixraven  thanks for that.

**Comment by Autofire** [comment]
Getting this error on version 0.49.0. I'm running Windows 10, and trying to run Ubuntu under WSL. My shell is Bash. I don't have a wsl.conf so that shouldn't be an issue.

It was working yesterday, but when I turned my computer on this morning, it no longer works. I'm getting this error message whenever I try to open a WSL tab. Cywgin tabs work, but the rest of the types are broken.

Note that I've only used WSL with ExtraTerm yesterday. Powershell and CMD may have been always broken.

WSL:
![image](https://user-images.githubusercontent.com/7464159/73772202-9280af80-4734-11ea-81f1-32a261d13dc1.png)

Powershell and CMD:
![image](https://user-images.githubusercontent.com/7464159/73774541-02913480-4739-11ea-8897-6dcc3a2ca789.png)
